normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
I lied. Dealing with the hash is more difficult because of the m2m relationships